modP2pImpl/src/org/aion/p2p/impl/comm/NodeMgr.java

```diff
@@ -204,22 +198,13 @@ public void addOutboundNode(final INode _n) {
     @Override
     public INode tempNodesTake() {
         INode node = null;
-        final ReentrantLock takeLock = this.takeLock;
+        tempNodesLock.lock();
         try {
-            takeLock.lockInterruptibly();
-            while (tempNodes.isEmpty()) {
-                notEmpty.await();
-            }
-
-            Integer key = tempNodes.keySet().iterator().next();
-            node = tempNodes.remove(key);
-            if (!tempNodes.isEmpty()) {
-                notEmpty.signal();
-            }
-        } catch (InterruptedException e) {
+            node = tempNodes.remove(tempNodes.keySet().iterator().next());
+        } catch (Exception e) {
             p2pLOG.error("<tempNodesTake IllegalStateException>", e);
```

Comment:
Log message is different from the actual exception being caught

Reply:
updated